crates/wasi-common/yanix/src/file.rs

```diff
@@ -48,14 +49,22 @@ bitflags! {
         const APPEND = libc::O_APPEND;
         const CREAT = libc::O_CREAT;
         const DIRECTORY = libc::O_DIRECTORY;
-        #[cfg(any(target_os = "android",
-                  target_os = "ios",
-                  target_os = "linux",
-                  target_os = "macos",
-                  target_os = "netbsd",
-                  target_os = "openbsd",
-                  target_os = "emscripten"))]
-        const DSYNC = libc::O_DSYNC;
+        const DSYNC = {
+            cfg_if! {
+                if #[cfg(any(target_os = "android",
+                             target_os = "ios",
+                             target_os = "linux",
+                             target_os = "macos",
+                             target_os = "netbsd",
+                             target_os = "openbsd",
+                             target_os = "emscripten"))] {
+                    libc::O_DSYNC
+                } else {
+                    // https://reviews.freebsd.org/D19407#inline-118670
+                    0
+                }
```

Comment:
Hmm, would it be possible to have `else if #[cfg(target_os = "freebsd")]` instead of an absorbing `else`? This way, on OSes we haven't considered yet would trip immediately in compile time which would save us from hunting potentially tricky bugs at runtime otherwise.